### PR TITLE
Fix: add ability to search order by uuid

### PR DESCRIPTION
**Pull request by akulakovaa**

### CODE REVIEW

File: app/api/v2/market/named_params.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ module NamedParams
 
         params :order_id do
           requires :id,
-                   type: { value: Integer, message: 'market.order.non_integer_id' },
+                   type: { value: String, message: 'market.order.non_integer_id' },
```
Comment (mnaichuk) [inline app/api/v2/market/named_params.rb]: change to `type: String`

File: app/api/v2/market/orders.rb [hunk 2]
```diff
@@ -79,8 +79,9 @@ class Orders < Grape::API
           use :order_id
         end
         get '/orders/:id' do
-          order = current_user.orders.find_by!(id: params[:id])
-          present order, with: API::V2::Entities::Order, type: :full
+          order = current_user.orders.find_by!(id: params[:id]) if (params[:id]).number?
```
Comment (mnaichuk) [inline app/api/v2/market/orders.rb]: it will break if the id, not int or correct uuid

File: app/api/v2/market/orders.rb [hunk 3]
```diff
@@ -79,8 +79,9 @@ class Orders < Grape::API
...
+          order = current_user.orders.find_by!(uuid: params[:id]) if UUID.validate(params[:id])
+        present order, with: API::V2::Entities::Order, type: :full
```
Comment (mnaichuk) [inline app/api/v2/market/orders.rb]: add 2 spaces here

File: config/initializers/string.rb [hunk 4]
```diff
@@ -8,4 +8,9 @@ def to_d
   rescue ArgumentError
     BigDecimal(0)
   end
+
+  def number?
+    return true if self =~ /\A[-+]?[0-9]+\z/
```
Comment (mnaichuk) [inline config/initializers/string.rb]: you can use `match?` without return true

File: lib/peatio/uuid.rb [hunk 5]
```diff
@@ -3,6 +3,9 @@ class << self
     def generate
       SecureRandom.uuid
     end
+    def validate(uuid)
+      return true if uuid =~ /\A[\da-f]{32}\z/i || uuid =~ /\A(urn:uuid:)?[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}\z/i
```
Comment (mnaichuk) [inline lib/peatio/uuid.rb]: same as in `number?`

File: spec/api/v2/market/orders_spec.rb [hunk 6]
```diff
@@ -194,6 +193,15 @@
       expect(result['executed_volume']).to eq '8.99'
     end
 
+    it 'should get specified order by uuid' do
```
Comment (mnaichuk) [inline spec/api/v2/market/orders_spec.rb:196]: add specs for invlaid uuid

File: spec/api/v2/market/orders_spec.rb [hunk 7]
```diff
@@ -203,13 +211,7 @@
       expect(result['trades'].first['id']).to eq trade.id
       expect(result['trades'].first['side']).to eq 'buy'
     end
-
-    it 'should get 404 error when order doesn\'t exist' do
```
Comment (mnaichuk) [inline spec/api/v2/market/orders_spec.rb:207]: you shouldn't delete this spec

**Review by mnaichuk — CHANGES_REQUESTED** [review]

**Review by mnaichuk — APPROVED** [review]

File: app/api/v2/market/orders.rb [hunk 8]
```diff
@@ -79,7 +79,13 @@ class Orders < Grape::API
           use :order_id
         end
         get '/orders/:id' do
-          order = current_user.orders.find_by!(id: params[:id])
+          if params[:id].number?
```
Comment (mod) [inline app/api/v2/market/orders.rb]: no need to extend string class
just test it here:  "12354".match?(/^[0-9]+$/)

File: spec/api/v2/market/orders_spec.rb (hunk 6 above)
Comment (mod) [reply spec/api/v2/market/orders_spec.rb:196]: I still don't see an example test
it would be nice to cover also lib/peatio/uuid.rb file with a spec file

**Review by mod — CHANGES_REQUESTED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**